code/game/gamemodes/blob/blobs/shield.dm

```diff
@@ -19,3 +19,32 @@
 /obj/structure/blob/shield/CanPass(atom/movable/mover, turf/target, height=0)
 	if(istype(mover) && mover.checkpass(PASSBLOB))	return 1
 	return 0
+
+/obj/structure/blob/shield/reflective
+	name = "reflective blob"
+	desc = "A solid wall of slightly twitching tendrils with a reflective glow."
+	icon_state = "blob_idle_glow"
+	brute_resist = 0
+	health = 50
+	maxHealth = 50
+	flags_1 = CHECK_RICOCHET_1
+	var/reflect_chance = 80 //80% chance to reflect
+
+/obj/structure/blob/shield/reflective/handle_ricochet(obj/item/projectile/P)
+	if(P.is_reflectable && prob(reflect_chance) && !istype(P, /obj/item/projectile/beam/pulse))

```

Comment:
Why an istype check? Shouldn't you just set pulse to be non-reflectable?

Reply:
Or just outright skip the change to pulse and make them reflectable, like they are with everything else